### PR TITLE
Improve/date library

### PR DESCRIPTION
**Pull request by 3lvis**
Finally fixed the stupid sun phases. Just went for figuring out how to use a JavaScript library from Swift and used https://github.com/mourner/suncalc. So now we show what http://daylight.today/app shows.

Fixes https://github.com/bakkenbaeck/daylight-ios/issues/50

### CODE REVIEW

File: Library/Location.swift [hunk 1]
```diff
@@ -60,25 +59,25 @@ struct Location {
     }
 
     var isNight: Bool {
-        return self.sunPhase == .night
+        return self.sunPhase == .nightEnd
```
Comment (marijnschilling) [inline Library/Location.swift]: Why is nightEnd the right phase for night? 🤔

File: Library/Location.swift [hunk 2]
```diff
@@ -60,25 +59,25 @@ struct Location {
...
-        let suntimes = Suntimes(date: Date(), timeZone: .autoupdatingCurrent, latitude: self.coordinate.latitude, longitude: self.coordinate.longitude)
+        let sunCalc = SunCalc(date: Date(), timeZone: TimeZone.autoupdatingCurrent, latitude: self.coordinate.latitude, longitude: self.coordinate.longitude)
 
-        return suntimes.sunPhase
+        return sunCalc.sunPhase
     }
 
     var daylightLengthProgress: Double {
-        let suntimes = Suntimes(date: Date(), timeZone: .autoupdatingCurrent, latitude: self.coordinate.latitude, longitude: self.coordinate.longitude)
+        let sunCalc = SunCalc(date: Date(), timeZone: TimeZone.autoupdatingCurrent, latitude: self.coordinate.latitude, longitude: self.coordinate.longitude)
```
Comment (marijnschilling) [inline Library/Location.swift]: Aaah so you added methods to our own `SunCalc` library or just renamed it?

File: Library/SunCalc.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import JavaScriptCore
+import Foundation
+
+enum SunPhase {
+    case nightStart
```
Comment (marijnschilling) [inline Library/SunCalc.swift]: Why is `nightStart` after `nightEnd`?

File: Library/SunCalc.swift [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    init(date: Date, timeZone: TimeZone, latitude: Double, longitude: Double, bundle: Bundle = Bundle.main) {
+        self.date = date
+        self.timeZone = timeZone
+
+        let path = bundle.path(forResource: "suncalc", ofType: "js")!
+        let contents = try! String(contentsOfFile: path)
+        let context = JSContext()!
+        context.evaluateScript(contents)
+
+        let function = context.objectForKeyedSubscript("getTimes")!
```
Comment (marijnschilling) [inline Library/SunCalc.swift]: Why is this called function? 😬 I'm not sure what this object is, maybe a better variable name would help

File: Library/SunCalc.swift [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    private(set) var sunrise: Date
+    private(set) var sunriseEnd: Date
+    private(set) var sunset: Date
+    private(set) var sunsetStart: Date
+
+    init(date: Date, timeZone: TimeZone, latitude: Double, longitude: Double, bundle: Bundle = Bundle.main) {
+        self.date = date
+        self.timeZone = timeZone
+
+        let path = bundle.path(forResource: "suncalc", ofType: "js")!
```
Comment (marijnschilling) [inline Library/SunCalc.swift]: What is this file exactly?

File: Library/SunCalc.swift (hunk 4 above)
Comment (marijnschilling) [inline Library/SunCalc.swift]: Aaah it's the `gtTimes` function in javaScript!

Maybe just calling it `getTimesWithJavaScriptFunction` ?

File: Library/SunCalc.swift [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        let contents = try! String(contentsOfFile: path)
+        let context = JSContext()!
+        context.evaluateScript(contents)
+
+        let function = context.objectForKeyedSubscript("getTimes")!
+        let times = function.call(withArguments: [date, latitude, longitude])!.toObjectOf(NSDictionary.self)! as! [String: Any]
+
+        self.dawn = times["dawn"] as? Date ?? Date()
```
Comment (marijnschilling) [inline Library/SunCalc.swift]: oke, so it gives the times of the sunphases on a certain date for a longitude/langitude.

I actually think some comments could be appropriate in this case. The case of using a javascript function in swift.

File: Library/Location.swift (hunk 1 above)
Comment (3lvis) [inline Library/Location.swift]: `.night` starts where a day starts, night end is basically the second time there's night that day.

File: Library/Location.swift (hunk 2 above)
Comment (3lvis) [inline Library/Location.swift]: Made my own based on https://github.com/mourner/suncalc/blob/master/suncalc.js

File: Library/SunCalc.swift (hunk 3 above)
Comment (3lvis) [inline Library/SunCalc.swift]: Night start is how a day begins

File: Library/SunCalc.swift (hunk 5 above)
Comment (3lvis) [inline Library/SunCalc.swift]: Added a few comments.

File: Library/SunCalc.swift (hunk 4 above)
Comment (3lvis) [inline Library/SunCalc.swift]: Renamed method.

File: Library/SunCalc.swift (hunk 6 above)
Comment (3lvis) [inline Library/SunCalc.swift]: Done! Added comments.

**Comment by 3lvis** [comment]
Implemented feedback, gonna merge this, if there's anything else, just let me know and I'll open an issue for it.

**Pull request merged**